third_party/util/atomic_writer.go

```diff
@@ -394,19 +395,31 @@ func (w *AtomicWriter) writePayloadToDir(payload map[string]FileProjection, dir
 			return err
 		}
 
-		if fileProjection.FsUser == nil {
+		// If neither FsUser or FsGroup defined, we don't need to Chown.
+		if fileProjection.FsUser == nil && fileProjection.FsGroup == nil {
 			continue
 		}
-		if err := os.Chown(fullPath, int(*fileProjection.FsUser), -1); err != nil {
-			klog.Errorf("%s: unable to change file %s with owner %v: %v", w.logContext, fullPath, int(*fileProjection.FsUser), err)
+
+		fsUser := -1
```

Comment:
what happens if we chown to -1? 🤔

Reply:
This means the [ownership doesn't change](https://golang.org/src/os/file_posix.go?s=2985:3028#L97). Will add a comment.